### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by Kobo-coder**
## Describe your changes

I've added some documentation for large overarching methods. Please roast me for my lack of explanation skills, as this is bound to be not great off the bat.

Also I've taken the liberty of making it so writing both `./byoqm` and `byoqm` will make no difference since, I myself found it a hassle to be forced to write `./<arbitrary folder name here>` and I thought it would be more ease of use.

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] Have I requested and notified the boys for review of the pull-request?

### CODE REVIEW

File: byoqm/dashboard/dashboard.py [hunk 1]
```diff
@@ -20,13 +20,12 @@ def _check_data(self, in_use_qm, target_path, filename):
             is_right_src = self._check_src_root(row[1], target_path)
             if not is_right_qm or not is_right_src:
                 return False
-
         return True
 
-    def _check_src_root(self, targetSrc, actualSrc):
+    def _check_src_root(self, actualSrc, targetSrc):
```
Comment (toffernator) [inline byoqm/dashboard/dashboard.py:25]: snake casing :eyes:

File: byoqm/runner/runner.py [hunk 2]
```diff
@@ -102,6 +107,14 @@ def _run_metrics(self) -> Dict:
         return results
 
     def _generate_violations_table(self, results: Dict, time: str):
+        """
+        _generate_violations_table generates a csv file in the output/violations/ folder containing
+        the locations of all found violations during parsing of the code base
+
+        _generate_violations_table generates given a dictionary of metrics, constructs a list containing
```
Comment (toffernator) [inline byoqm/runner/runner.py:114]: The middle step adds confusion without clarification (it makes a list to then make a pandas dataframe). To me the important aspect here is I give a dict and I get a file

File: byoqm/source_repository/source_repository.py [hunk 3]
```diff
@@ -8,6 +8,10 @@
 
 
 class SourceRepository:
+    """
+    represents a repository containing all information accessible by metrics
```
Comment (toffernator) [inline byoqm/source_repository/source_repository.py:12]: We are not using repository to refer to a code base, but rather to refer to the repository architecture.
```suggestion
    contains all information accessible by metrics
```

**Review by FrederikRothe — APPROVED** [review]
Looks good, no comments ✅

File: byoqm/dashboard/test_dashboard.py [hunk 4]
```diff
@@ -78,7 +78,7 @@ def test_get_data_given_1_day_seperation_returns_2(self):
         data = dashboard.get_data(
             path=_OUTPUT_FOLDER,
             in_use_qm="dummyqm",
-            targetPath="./dummysrc",
+            targetPath=Path("./dummysrc"),
```
Comment (SadCheeto) [inline byoqm/dashboard/test_dashboard.py:81]: Should this not be snake casing as well?

File: byoqm/dashboard/test_dashboard.py [hunk 5]
```diff
@@ -65,7 +65,7 @@ def test_get_data_given_2023_returns_4(self):
...
```
Comment (SadCheeto) [inline byoqm/dashboard/test_dashboard.py:68]: Also snake casing no?

**Review by SadCheeto — APPROVED** [review]
Looks good overall. Spotted some camel casing, but other than that we can just merge it

**Pull request merged**